main.c

```diff
@@ -459,11 +460,51 @@ static void dbus_name_lost(GDBusConnection *connection,
 	tcmu_dbg("name lost\n");
 }
 
+static const char *dirname_default_prefix = "/lib/modules";
+
+static char *get_modules_dirpath(void)
+{
+	struct utsname u;
+	char *p;
+
+	if (uname(&u) < 0)
+		return NULL;
+
+	if (asprintf(&p, "%s/%s", dirname_default_prefix, u.release) < 0)
+		return NULL;
+
+	return p;
+}
+
 static int load_our_module(void)
 {
 	struct kmod_list *list = NULL, *itr;
 	struct kmod_ctx *ctx;
 	int ret;
+	struct stat sb;
+	char *modules_dir;
+
+	modules_dir = get_modules_dirpath();
+	if (!modules_dir) {
+		tcmu_err("get_modules_dirpath() failed\n");
+		return -1;
+	}
+
+	/* check if path entry exist */
+	if (stat(modules_dir, &sb))
+	{
+		free(modules_dir);
```

Comment:
If you're not going to log the modules dir in the failure log message just free it once when your done with it:

ret = stat()
free(modules_dir);
if (ret) {
 ....
}